runtime/lua/man.lua

```diff
@@ -0,0 +1,141 @@
+local function highlight_formatted(line, linenr)
+  local chars = {}
+  local prev_char = ''
+  local overstrike, escape = false, false
+  local hls = {} -- Store highlight groups as { attr, start, end }
```

Comment:
This is really janky imo to not name the fields stored in that tuple but I'm not sure if there is a better way. I have very little lua experience.

Reply:
Yeah, I'll name them